docs/essentials/widget_language.md

```diff
@@ -4,15 +4,15 @@ sidebar_position: 1
 
 # The `widget!` macro
 
-`widget!` macro is a declarative language to help you to build your declare and reactive UI  quickly and expressively. It allows user interfaces to be described in terms of their visual struct, and almost all its syntaxes are expanded from the syntax of Rusts. So don't worry about the learning curve of this DSL language.
+The `widget!` macro is a declarative language to help you to build and declare your reactive UI quickly and expressively. It allows user interfaces to be described in terms of their visual structure, and almost all its syntaxes are expanded to the syntax of Rusts. So don't worry about the learning curve of this DSL language.
```

Comment:
Look at the sentence: "and almost all its syntaxes are expanded to the syntax of Rusts"：
" are expanded to" may not be precise.  I would like to use another phrasal verb: derives from; Do you think it is better?

Reply:
I'm not sure about that. I think `syntaxes` should be changed to singular though.
Most people trying Ribir probably aren't Rust newbies and they have an idea about what macros do already. So I wouldn't worry too much about it.

Reply:
Alright~